### PR TITLE
Adding test 'test_many_cores_raid_as_cache'

### PR DESCRIPTION
**Pull request by karolinavelkaja**
Signed-off-by: Karolina Rogowska <karolina.rogowsky@gmail.com>

### CODE REVIEW

File: test/functional/tests/fault_injection/test_fault_injection_many_cores_raid_as_cache.py [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+        test_file = fs_utils.create_random_test_file(temp_file, test_file_size)
+        test_file.copy(destination_file, force=True)
+
+        md5sums.append(test_file.md5sum())
+
+    TestRun.LOGGER.info(f"Files created and copied to core successfully.")
+    return md5sums
+
+
+def compare_md5sums(md5_sums_source, files_cto_check_path):
```
Comment (Ostrokrzew) [inline test/functional/tests/fault_injection/test_fault_injection_many_cores_raid_as_cache.py:105]: 'cto' in argument name
I assume it's typo, if not, sorry.

File: test/functional/tests/fault_injection/test_fault_injection_many_cores_raid_as_cache.py [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+    md5_sums_elements = len(md5_sums_source)
+
+    for i in range(0, md5_sums_elements):
```
Comment (Ostrokrzew) [inline test/functional/tests/fault_injection/test_fault_injection_many_cores_raid_as_cache.py:108]: This '0' isn't necessary.

File: test/functional/tests/fault_injection/test_fault_injection_many_cores_raid_as_cache.py [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+        core2_md5sums = create_files(mount_point2, number_of_files)
+
+    with TestRun.step("Compare checksum on first core."):
+        compare_md5sums(core1_md5sums, mount_point)
+
+    with TestRun.step("Compare checksum on second core."):
+        compare_md5sums(core2_md5sums, mount_point2)
+
+
+def create_files(destination_path, files_count):
```
Comment (Ostrokrzew) [inline test/functional/tests/fault_injection/test_fault_injection_many_cores_raid_as_cache.py:90]: I suggest You to rename this function, now it's not fully descriptive.

**Review by Ostrokrzew — CHANGES_REQUESTED** [review]

**Review by Ostrokrzew — DISMISSED** [review]

File: test/functional/tests/fault_injection/test_fault_injection_many_cores_raid_as_cache.py [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+
+        raid_volume = Raid.create(config, [raid_disk_1, raid_disk_2])
+        TestRun.LOGGER.info(f"RAID created successfully.")
+
+    with TestRun.step("Add core device to cache."):
+        core_disk1 = TestRun.disks['core1']
+        core_disk1.create_partitions([Size(2, Unit.GibiByte)])
+        core_dev1 = core_disk1.partitions[0]
+
+        cache = casadm.start_cache(raid_volume, cache_mode, force=True)
```
Comment (katlapinka) [inline test/functional/tests/fault_injection/test_fault_injection_many_cores_raid_as_cache.py:62]: this line should be placed in the previous step

File: test/functional/tests/fault_injection/test_fault_injection_many_cores_raid_as_cache.py [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+        config = RaidConfiguration(
+            level=Level.Raid0,
+            metadata=MetadataVariant.Legacy,
+            number_of_devices=2,
+            size=Size(1, Unit.GiB))
+
+        raid_volume = Raid.create(config, [raid_disk_1, raid_disk_2])
+        TestRun.LOGGER.info(f"RAID created successfully.")
+
+    with TestRun.step("Add core device to cache."):
```
Comment (katlapinka) [inline test/functional/tests/fault_injection/test_fault_injection_many_cores_raid_as_cache.py:57]: "... , create filesystem and mount it"

File: test/functional/tests/fault_injection/test_fault_injection_many_cores_raid_as_cache.py [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+        core_disk1.create_partitions([Size(2, Unit.GibiByte)])
+        core_dev1 = core_disk1.partitions[0]
+
+        cache = casadm.start_cache(raid_volume, cache_mode, force=True)
+        core1 = cache.add_core(core_dev1)
+
+        core1.create_filesystem(Filesystem.ext3)
+        core1.mount(mount_point)
+
+    with TestRun.step("Add second core device to cache."):
```
Comment (katlapinka) [inline test/functional/tests/fault_injection/test_fault_injection_many_cores_raid_as_cache.py:68]: "... , create filesystem and mount it"

**Review by katlapinka — CHANGES_REQUESTED** [review]

**Review by Ostrokrzew — APPROVED** [review]

File: test/functional/tests/fault_injection/test_fault_injection_many_cores_raid_as_cache.py [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+
+
+@pytest.mark.parametrize("cache_mode", CacheMode)
+@pytest.mark.require_disk("cache1", DiskTypeSet([DiskType.optane, DiskType.nand]))
+@pytest.mark.require_disk("cache2", DiskTypeSet([DiskType.optane, DiskType.nand]))
+@pytest.mark.require_disk("core1", DiskTypeLowerThan("cache1"))
+@pytest.mark.require_disk("core2", DiskTypeLowerThan("cache1"))
+def test_fault_injection_many_cores_raid_as_cache(cache_mode):
+    """
+        title: Test if CAS is working with many cores devices using RAID0 as cache device.
```
Comment (katlapinka) [inline test/functional/tests/fault_injection/test_fault_injection_many_cores_raid_as_cache.py:30]: "many core devices" - without "s" at the end of "cores"

File: test/functional/tests/fault_injection/test_fault_injection_many_cores_raid_as_cache.py [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+        description: |
+          Test if CAS is working properly with many cores devices using RAID0 as cache device
```
Comment (katlapinka) [inline test/functional/tests/fault_injection/test_fault_injection_many_cores_raid_as_cache.py:32]: here as well :)

**Review by Ostrokrzew — DISMISSED** [review]

File: test/functional/tests/volumes/test_many_cores_raid_as_cache.py [hunk 9]
```diff
@@ -0,0 +1,115 @@
...
+@pytest.mark.require_disk("cache1", DiskTypeSet([DiskType.optane, DiskType.nand]))
+@pytest.mark.require_disk("cache2", DiskTypeSet([DiskType.optane, DiskType.nand]))
+@pytest.mark.require_disk("core1", DiskTypeLowerThan("cache1"))
+@pytest.mark.require_disk("core2", DiskTypeLowerThan("cache1"))
+def test_many_cores_raid_as_cache(cache_mode):
+    """
+        title: Test if CAS is working with many core devices using RAID0 as cache device.
+        description: |
+          Test if CAS is working properly with many core devices using RAID0 as cache device
+           and verification of data integrity of files copied to cores.
```
Comment (katlapinka) [inline test/functional/tests/volumes/test_many_cores_raid_as_cache.py:33]: In my opinion  the first line of the description is sufficient :)

File: test/functional/tests/volumes/test_many_cores_raid_as_cache.py [hunk 10]
```diff
@@ -0,0 +1,115 @@
...
+        raid_disk_1 = raid_disk.partitions[0]
+        raid_disk2 = TestRun.disks['cache2']
+        raid_disk2.create_partitions([Size(2, Unit.GibiByte)])
+        raid_disk_2 = raid_disk2.partitions[0]
+
+        config = RaidConfiguration(
+            level=Level.Raid0,
+            metadata=MetadataVariant.Legacy,
+            number_of_devices=2,
+            size=Size(1, Unit.GiB))
```
Comment (katlapinka) [inline test/functional/tests/volumes/test_many_cores_raid_as_cache.py:52]: Size unnecessary

**Review by jfckm — APPROVED** [review]

**Review by robertbaldyga — APPROVED** [review]

**Pull request merged**